pkg/ddl/backfilling_import_cloud.go

```diff
@@ -78,10 +78,31 @@ func (m *cloudImportExecutor) RunSubtask(ctx context.Context, subtask *proto.Sub
 	if local == nil {
 		return errors.Errorf("local backend not found")
 	}
-	// TODO(lance6716): find the correct index, so "duplicate entry" error can have index name
-	currentIdx := m.indexes[0]
 
-	_, engineUUID := backend.MakeUUID(m.ptbl.Meta().Name.L, currentIdx.ID)
+	var (
+		currentIdx *model.IndexInfo
+		idxID      int64
+	)
+	switch len(sm.EleIDs) {
+	case 1:
+		for _, idx := range m.indexes {
+			if idx.ID == sm.EleIDs[0] {
+				currentIdx = idx
+				idxID = idx.ID
+				break
+			}
+		}
+	case 0:
```

Comment:
maybe we should write in doc that we don't support using global sort during upgrade, so we don't need to maintain this in-compatibility

Reply:
From a user perspective, it would be best if we could accommodate this case and not block users from upgrading. Otherwise, a lot of restrictions would hinder the promotion and usage of this feature. Therefore, I suggest we optimize internally and avoid blocking upgrades.